ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -571,7 +552,15 @@ private EndpointResponse handleStreamPullQuery(
       final ImmutableAnalysis analysis,
       final ServiceContext serviceContext,
       final ConfiguredStatement<Query> configured,
-      final CompletableFuture<Void> connectionClosedFuture) {
+      final CompletableFuture<Void> connectionClosedFuture,
+      final AtomicReference<StreamPullQueryMetadata> resultForMetrics,
+      final AtomicReference<Optional<Decrementer>> optionalDecrementer) {
+
+    // Apply the same rate, bandwidth and concurrency limits as with table pull queries
+    PullQueryExecutionUtil.checkRateLimit(rateLimiter);
+    final Decrementer decrementer = concurrencyLimiter.increment();
+    pullBandRateLimiter.allow(KsqlQueryType.PULL);
+    optionalDecrementer.set(Optional.ofNullable(decrementer));
```

Comment:
Looking at the ConcurrencyLimiter implementation, I don't believe `decrementer` actually could ever be null.

More generally, it doesn't look like we really need to have an `AtomicReference<Optional<Decrementer>>`. It seems like just an `AtomicReference<Decrementer>` would do just fine.

The reason is that the AtomicReferenced starts off initialized to null, which means the referencer needs to check for null anyway, so the Optional really isn't buying us anything here.